nifi-docs/src/main/asciidoc/toolkit-guide.adoc

```diff
@@ -0,0 +1,1257 @@
+
+==== Restore
+The restore operation allows an existing installation to revert back to a previous installation.  Using an existing backup directory (created from the backup operation) the FileManager utility will restore libraries, scripts and documents as well as revert to previous configurations.
+
+NOTE: If repositories were changed due to the installation of a newer version of NiFi these may no longer be compatible during restore.  In that scenario exclude the `-m` option to ensure new repositories will be created or, if repositories live outside of the NiFi directory, remove them so they can be recreated on startup after restore.
+
+== Flow Analyzer
+The `flow-analyzer` tool (invoked as `./bin/flow-analyzer.sh` or `bin\flow-analyzer.bat`) analyzes the _flow.xml.gz_ file and reports:
+
+* Total Bytes Utilized by the System
+* Min/Max Back Pressure Size
+* Average Back Pressure Size
+* Min/Max Flowfile Queue Size
+* Average Flowfile Queue Size
+
+=== Usage
+To execute the `flow-analyzer` tool:
+
+ flow-analyzer.sh <path to flow.xml.gz>
+
+Example:
+
+ $ ./flow-analyzer.sh /Users/nifiuser/nifi-1.8.0/conf/flow.xml.gz
+ Using flow=/Users/nifiuser/nifi-1.8.0/conf/flow.xml.gz
+ Total Bytes Utilized by System=1518 GB
+ Max Back Pressure Size=1 GB
+ Min Back Pressure Size=1 GB
+ Average Back Pressure Size=2.504950495 GB
+ Max Flowfile Queue Size=10000
+ Min Flowfile Queue Size=10000
+ Avg Flowfile Queue Size=10000.000000000
+
+== Node Manager
+Node manager (invoked as `./bin/node-manager.sh` or `bin\node-manager.bat`) supports connecting, disconnecting and removing a node when in a cluster (an error message displays if the node is not part of a cluster) as well as obtaining the status of a node.  When nodes are disconnected from a cluster and need to be connected or removed, a list of urls of connected nodes should be provided to send the required command to the active cluster.  Node Manager supports NiFi version 1.0.0 and higher.
+
+=== Usage
+To show help:
+
+  node-manager.sh -h
+
+The following are available options:
+
+* `-b`,`--bootstrapConf <arg>`     Existing Bootstrap Configuration file (required)
+* `-d`,`--nifiInstallDir <arg>`    NiFi Root Folder (required)
+* `-h`,`--help`                    Help Text (optional)
+* `-o`, `--operation <arg>`        Operations supported: status, connect (cluster), disconnect (cluster), remove (cluster)
+* `-p`,`--proxyDN <arg>`           Proxy or User DN (required for secured nodes doing connect, disconnect and remove operations)
+* `-u`,`--clusterUrls <arg>`       Comma delimited list of active urls for cluster (optional). Not required for disconnecting a node yet will be needed when connecting or removing from a cluster
+* `-v`,`--verbose`                 Verbose messaging (optional)
+
+
+To connect, disconnect, or remove a node from a cluster:
+
+ node-manager.sh -d {$NIFI_HOME} –b { nifi bootstrap file path}
+ -o {remove|disconnect|connect|status} [-u {url list}] [-p {proxy name}] [-v]
```

Comment:
Not related to the PR itself but wondering if it'd make sense to add 'offload' operation (cc @jtstorck) even though it's available in the CLI.